client/blocks/site-preview/index.jsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import WebPreview from 'components/web-preview';
+import designPreview from 'my-sites/design-preview';
+import urlPreview from 'blocks/url-preview';
+import { getCurrentLayoutFocus } from 'state/ui/layout-focus/selectors';
+import { getCurrentPreviewType } from 'state/ui/preview/selectors';
+
+class SitePreview extends Component {
+	constructor( props ) {
+		super( props );
+		this.getPreviewComponent = this.getPreviewComponent.bind( this );
+	}
+
+	getPreviewComponent() {
+		switch ( this.props.currentPreviewType ) {
+			case 'design-preview':
+				const DesignPreview = designPreview( WebPreview );
+				return <DesignPreview showPreview={ this.props.showPreview } />;
+			case 'site-preview':
+				const UrlPreview = urlPreview( WebPreview );
+				return <UrlPreview showPreview={ this.props.showPreview } />;
+		}
+		return null;
```

Comment:
looks like we're doing the same operation (adding `showPreview`) to a bunch of different components. We could make the component static…

```js
getPreviewComponent() {
	const { currentPreviewType, showPreview } = this.props;

	const Component = get( [
		'design-preview': designPreview( WebPreview ),
		'site-preview': urlPreview( WebPreview )
	], currentPreviewType, () => null )

	return React.cloneElement( Component, { showPreview } );
}
```

…just an idea which could prevent some redundant code if we add more options